### PR TITLE
fix(rust, python): Allow initialize of pl.Array in Dataframe using schema alone

### LINKED ISSUE #10094: Cannot initialize empty dataframe from schema containing Array dtype
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl
schema = {
    "foo": pl.Int64,
    "bar": pl.Struct({"a":pl.Int64, "b":pl.Float32, "c":pl.Utf8}),
     "baz": pl.Array(width=3, inner=pl.Int32), # this one fails
}
pl.DataFrame(schema=schema)
```
```
ValueError: could not create Array from input
```

### Issue description

If you pass a schema dict to pl.DataFrame without passing `data`, you can normally create an empty DataFrame:
```python
import polars as pl
schema = {
    "foo": pl.Int64,
    "bar": pl.Struct({"a":pl.Int64, "b":pl.Float32, "c":pl.Utf8}),
}
pl.DataFrame(schema=schema)
```
But if you add a column with dtype `pl.Array` to the schema, this fails, as per above.

It works if you do pass a `data` dict along it:

```python
import polars as pl

schema = {
    "foo": pl.Int64,
    "bar": pl.Struct({"a": pl.Int64, "b": pl.Float32, "c": pl.Utf8}),
    "baz": pl.Array(width=3, inner=pl.Int32),  # this one fails
}
pl.DataFrame(
    {
        "foo": [1],
        "bar": [{"a": 1, "b": 2.0, "c": "abc"}],
        "baz": [[1, 2, 3]]
    },
    schema=schema,
)
``` 

### Expected behavior

The same behaviour across all dtypes

### Installed versions

<details>

```
--------Version info---------
Polars:              0.18.8
Index type:          UInt32
Platform:            Linux-5.15.49-linuxkit-pr-aarch64-with-glibc2.31
Python:              3.10.11 (main, May 23 2023, 09:27:52) [GCC 10.2.1 20210110]

----Optional dependencies----
adbc_driver_sqlite:  0.5.1
cloudpickle:         2.2.1
connectorx:          <not installed>
deltalake:           0.10.0
fsspec:              2023.6.0
matplotlib:          3.7.2
numpy:               1.25.1
pandas:              2.0.3
pyarrow:             10.0.1
pydantic:            1.10.11
sqlalchemy:          1.4.49
xlsx2csv:            0.8.1
xlsxwriter:          3.1.2
```

</details>

### PR DESCRIPTION
**Pull request by mcrumiller**
Resolves #10094

### CODE REVIEW

File: py-polars/src/series/construction.rs [hunk 1]
```diff
@@ -233,17 +233,28 @@ impl PySeries {
...
-                let out = out
-                    .cast(&DataType::Array(
-                        Box::new(inner.map(|dt| dt.0).unwrap_or(*list_inner.clone())),
-                        width,
-                    ))
-                    .map_err(PyPolarsErr::from)?;
-                Ok(out.into())
+        if out.len() == 0 {
+            // no inner list present
+            let out = out
```
Comment (mcrumiller) [inline py-polars/src/series/construction.rs:238]: @ritchie46 I'm not sure if there's a more straightforward way of creating an empty Series with a specified Array dtype.

Comment (ritchie46) [reply py-polars/src/series/construction.rs:238]: Yes, you should use `Series::new_empty()` in that case. :)

File: py-polars/src/series/construction.rs [hunk 2]
```diff
@@ -233,17 +233,28 @@ impl PySeries {
...
-        match out.dtype() {
-            DataType::List(list_inner) => {
-                let out = out
-                    .cast(&DataType::Array(
-                        Box::new(inner.map(|dt| dt.0).unwrap_or(*list_inner.clone())),
-                        width,
-                    ))
-                    .map_err(PyPolarsErr::from)?;
-                Ok(out.into())
+        if out.len() == 0 {
```
Comment (ritchie46) [inline py-polars/src/series/construction.rs:236]: Let's make this branch `if val.is_empty` and move the lines 

```rust
  let val = vec_extract_wrapped(val);
  let out = Series::new(name, &val);
```

into the second branch.

File: py-polars/src/series/construction.rs (hunk 1 above)
Comment (mcrumiller) [reply py-polars/src/series/construction.rs]: done.

File: py-polars/src/series/construction.rs [hunk 3]
```diff
@@ -232,18 +232,24 @@ impl PySeries {
         _strict: bool,
     ) -> PyResult<Self> {
         let val = vec_extract_wrapped(val);
```
Comment (ritchie46) [inline py-polars/src/series/construction.rs:234]: This `val` should also in the second branch. We don't have to execute `val_extract_wrapped` if it is empty.

**Review by ritchie46 — COMMENTED** [review]
Almost there!

File: py-polars/src/series/construction.rs (hunk 3 above)
Comment (mcrumiller) [reply py-polars/src/series/construction.rs:234]: Can we call `is_empty()` directly on a wrapped val or do I need to do `val.0.is_empty()`?

Comment (mcrumiller) [reply py-polars/src/series/construction.rs:234]: Compiler answered my question.

Comment (mcrumiller) [reply py-polars/src/series/construction.rs]: Ok done.

**Pull request merged**